internal/services/springcloud/spring_cloud_java_deployment_resource.go

```diff
@@ -77,21 +96,44 @@ func resourceSpringCloudJavaDeployment() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			// TODO: Remove in 3.0
+			// The value returned in GET will be recalculated by the service if "memory_v2" is honored, so make this property as Computed.
 			"memory_in_gb": {
-				Type:         pluginsdk.TypeInt,
-				Optional:     true,
-				Default:      1,
-				ValidateFunc: validation.IntBetween(1, 8),
+				Type:          pluginsdk.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ValidateFunc:  validation.IntBetween(1, 8),
+				ConflictsWith: []string{"memory_v2"},
+				Deprecated:    "This field has been deprecated in favour of `memory_v2` and will be removed in a future version of the provider",
+			},
+
+			// The value returned in GET will be recalculated by the service if "memory_in_gb" is honored, so make this property as Computed.
+			"memory_v2": {
```

Comment:
we can just use this 
```suggestion
			"memory_in_mb": {
```

and then the user can specify it either in MB or GB allowing the half gig with this options